spec/outputs/elasticsearch_spec.rb

```diff
@@ -306,6 +306,56 @@
     end
   end
 
+  describe "http client create" do
+    require "logstash/outputs/elasticsearch"
+    require "elasticsearch"
+
+    let(:options) {
+      {
+        "protocol" => "http",
+        "index" => "my-index",
+        "host" => "localhost",
+        "path" => "some-path"
+      }
+    }
+
+    let(:eso) {LogStash::Outputs::ElasticSearch.new(options)}
+
+    let(:manticore_host) {
+      eso.client.first.send(:client).transport.options[:host].first
+    }
+
+    around(:each) do |block|
+      thread = eso.register
```

Comment:
It's returning a thread because of ruby's implicit return _and_ it was written long before @colinsurprenant convinced me that implicit return was good :P  -- my old assertion was that "methods that should return things should use the `return` statement" which makes any return value from the register method, under this idea, undefined/unsupported, even if accidental.

Best guess, for now, remove the `around(...) ...` block and the threads will die when rspec exits. We can refactor later for easier thread management once we know what/how things are interruptable.